llvm/lib/Target/EraVM/EraVMPostCodegenPrepare.cpp

```diff
@@ -0,0 +1,128 @@
+//===-- EraVMPostCodegenPrepare.cpp - EraVM Post CodeGenPrepare -*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This pass performs optimizations after CodeGenPrepare pass. Currently, it is
```

Comment:
Does "currently" imply that we plan to perform all signed to unsigned comparisons in this pass one day?

Reply:
Currently in this context means we want to use it for other optimizations, and not only for this one. Here is my suggestion what we can do:
1. Move handling of call instruction from `EraVMCodegenPrepare`.
2. Rename `EraVMCodegenPrepare` to something like `EraVMConvertPointerArithmetics`.
3. Rename `EraVMPostCodegenPrepare` to `EraVMCodegenPrepare`.
4. Implement `rearrangeOverflowHandlingBranches` from [here](https://github.com/matter-labs/era-compiler-llvm/commit/f31aa85caee7ccb207d66807cb6792cb7b8fa60b#diff-9b0cab2bf4202033e73cf5e22c78e0d1d72186d157542fca76d292a04985d637) to this pass, since CodeGenPrepare can also generate overflow intrinsics.

If you are OK with this, I will create add TODOs for these items.

Reply:
Yes, please

Reply:
Done. Created ticket for renaming, and added ticket numbers in the code.